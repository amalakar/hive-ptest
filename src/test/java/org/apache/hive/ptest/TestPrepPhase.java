/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.hive.ptest;

import java.io.File;
import java.util.concurrent.Executors;

import org.apache.commons.io.FileUtils;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.JunitReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;


@UseReporter(JunitReporter.class)
public class TestPrepPhase extends AbstractTestPhase {
  private PrepPhase phase;
  private File baseDir;
  
  @Before
  public void setup() throws Exception {
    super.setup();
    baseDir = createBaseDir(this.getClass().getSimpleName());
  }
  @After
  public void teardown() {
    FileUtils.deleteQuietly(baseDir);
  }
  @Test
  public void testExecute() throws Exception {
    phase = new PrepPhase(drones, executor, sshCommandExecutor, 
        scpCommandExecutor, rsyncCommandExecutor, localCommandFactory, 
        templateDefaults, baseDir, (File)null, Executors.newSingleThreadExecutor());
    phase.execute();
    Approvals.verify(getExecutedCommands());
  }
}
