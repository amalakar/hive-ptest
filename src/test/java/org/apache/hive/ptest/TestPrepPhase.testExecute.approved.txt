mkdir -p /some/local/dir/somehost-someuser-13/logs /some/local/dir/somehost-someuser-13/maven /some/local/dir/somehost-someuser-13/scratch
mkdir -p /some/local/dir/somehost-someuser-13/ivy /some/local/dir/somehost-someuser-13/repositoryName-source
mkdir -p /some/working/dir/scratch /some/working/dir/isolated/logs /some/working/dir/isolated/maven
mkdir -p /some/working/dir/isolated/scratch /some/working/dir/isolated/ivy /some/working/dir/isolated/repositoryName-source
bash /tmp/hive-ptest-units/TestPrepPhase/source-prep.sh
rsync -qavP --delete --delete-during --force /some/working/dir/repositoryName-source /some/working/dir/isolated/
rsync -qavP --delete --delete-during --force /some/working/dir/ivy /some/working/dir/isolated/
rsync -qavP --delete --delete-during --force /some/working/dir/maven /some/working/dir/isolated/
/some/working/dir/repositoryName-source /some/local/dir/somehost-someuser-13/
/some/working/dir/maven /some/local/dir/somehost-someuser-13/
/some/working/dir/ivy /some/local/dir/somehost-someuser-13/